Makefile

```diff
@@ -1,82 +1,94 @@
-	docker build -t tink-worker cmd/tink-worker/
-virtual-worker-image: cmd/virtual-worker/virtual-worker-linux-amd64
-	docker build -t virtual-worker cmd/virtual-worker/
-
-ifeq ($(origin GOBIN), undefined)
-GOBIN := ${PWD}/bin
-export GOBIN
-PATH := ${GOBIN}:${PATH}
-export PATH
-endif
+# Installed tools
+PROTOC_GEN_GO_GRPC 	:= google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.2.0
+PROTOC_GEN_GO 		:= google.golang.org/protobuf/cmd/protoc-gen-go@v1.28.1
 
-# These need changing to a `go run` command and placed in the Tools section of the Makefile
-.PHONY: tools ## Install all required tools for use with this Makefile.
-tools:
-	$(GO) install google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.2.0
-	$(GO) install google.golang.org/protobuf/cmd/protoc-gen-go@v1.28.1
-	$(GO) install sigs.k8s.io/controller-runtime/tools/setup-envtest@v0.0.0-20220304125252-9ee63fc65a97
-	$(GO) install sigs.k8s.io/kustomize/kustomize/v4@v4.5.4
+.PHONY: help
+help: ## Print this help
+	@awk 'BEGIN {FS = ":.*##"; printf "Usage:\n  make \033[36m<target>\033[0m\n"} /^[%\/0-9A-Za-z_-]+:.*?##/ { printf "  \033[36m%-20s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+	@echo
+	@echo Individual binaries can be built with their name. For example, \`make tink-server\`.
+	@echo
+	@echo Individual images can be built with their name appended with -image. For example, 
+	@echo   \`make tink-server-image\`.
+
+# Define all the binaries we build for this project that get packaged into containers.
+BINARIES := tink-server tink-worker tink-controller virtual-worker
+
+# Version defines the string injected into binaries that indicates the version of the build.
+# Eventually this will be superfluous as we transition to buildinfo provided by the Go standard
+# library.
+# Override by setting VERSION when invoking make. For example, `make VERSION=1.2.3`.
+VERSION ?= $(shell git rev-parse --short HEAD)
+
+.PHONY: build
+build: $(BINARIES) ## Build all tink binaries. Cross build by setting GOOS and GOARCH.
+
+# Create targets for all the binaries we build. They can be individually invoked with `make <binary>`.
+# For example, `make tink-server`. Callers can cross build by defining the GOOS and GOARCH 
+# variables. For example, `GOOS=linux GOARCH=arm64 make tink-server`.
+# See https://www.gnu.org/software/make/manual/html_node/Automatic-Variables.html.
+.PHONY: $(BINARIES)
+$(BINARIES):
+	CGO_ENABLED=false \
+	GOOS=$(GOOS) \
+	GOARCH=$(GOARCH) \
+	$(GO) build \
+		$(LDFLAGS) \
+		-o ./bin/$@-$(GOOS)-$(GOARCH) \
+		./cmd/$@
+
+# IMAGE_ARGS is resolved when its used in the `%-image` targets. Consequently, the $* automatic 
+# variable isn't evaluated until the target is called.  
+# See https://www.gnu.org/software/make/manual/html_node/Automatic-Variables.html.
+IMAGE_ARGS ?= -t $*
+
+.PHONY: images
+images: $(addsuffix -image, $(BINARIES)) ## Build all tink container images. All images will be for Linux but target the host arch.
```

Comment:
Hmm....this target is not working for me. `make images`: `make: Nothing to be done for 'images'`
same with the individual image targets. `make: Nothing to be done for 'tink-server-image'`

Reply:
fixed, thanks!